app/react/kubernetes/applications/event.service.ts

```diff
@@ -0,0 +1,44 @@
+import { EventList } from 'kubernetes-types/core/v1';
+import { useQuery } from 'react-query';
+
+import { EnvironmentId } from '@/react/portainer/environments/types';
+import axios, { parseAxiosError } from '@/portainer/services/axios';
+import { withError } from '@/react-tools/react-query';
+
+async function getNamespaceEvents(
+  environmentId: EnvironmentId,
+  namespace: string,
+  labelSelector?: string
+) {
+  try {
+    const { data } = await axios.get<EventList>(
+      `/endpoints/${environmentId}/kubernetes/api/v1/namespaces/${namespace}/events`,
+      {
+        params: {
+          labelSelector,
+        },
+      }
+    );
+    return data.items;
+  } catch (e) {
+    throw parseAxiosError(e as Error, 'Unable to retrieve events');
+  }
+}
+
+export function useNamespaceEventsQuery(
```

Comment:
this file should be called `useNamespaceEventsQuery` instead of `event.service` and should be either under `applications/queries` or in the same path as the table (if that's the only user for this function).